node/src/reactor/main_reactor/keep_up.rs

```diff
@@ -616,32 +615,42 @@ impl MainReactor {
         // on an old block. in either case we will attempt to get the next needed block (if any).
         // note: for a synced historical block we have header, body, global state, any execution
         // effects, any referenced deploys, & sufficient finality (by weight) of signatures.
-        let highest_orphaned_block_header = self.storage.get_highest_orphaned_block_header();
-        match highest_orphaned_block_header {
-            HighestOrphanedBlockResult::Orphan(block_header) => {
+        match self.storage.get_highest_orphaned_block_header() {
+            HighestOrphanedBlockResult::Orphan(highest_orphaned_block_header) => {
                 // set a latch on the validator matrix to prevent it from purging validator weights
                 // of interstitial eras which we have not yet historically sync'd
                 self.validator_matrix
-                    .register_retrograde_latch(Some(block_header.era_id()));
-                if block_header.is_genesis() {
+                    .register_retrograde_latch(Some(highest_orphaned_block_header.era_id()));
+                if highest_orphaned_block_header.is_genesis() {
                     return Ok(Some(SyncBackInstruction::GenesisSynced));
                 }
-                // if sync to genesis is false, we require sync to ttl; i.e. if the TTL is 12 hours
-                // we require sync back to see a contiguous / unbroken range of at least 12 hours
-                // worth of blocks. note however that we measure from the start of the active era
+
+                // if sync to genesis is false, we require sync to ttl; i.e. if the TTL is 12
+                // hours we require sync back to see a contiguous / unbroken
+                // range of at least 12 hours worth of blocks. note however
+                // that we measure from the start of the active era
                 // (for consensus reasons), so this can be up to TTL + era length in practice
-                if !self.sync_to_genesis
-                    && synced_to_ttl(
-                        None,
-                        Some(&block_header),
-                        &self.storage,
-                        self.chainspec.deploy_config.max_ttl,
-                    )?
-                {
-                    return Ok(Some(SyncBackInstruction::TtlSynced));
+                if !self.sync_to_genesis {
+                    if let Some(highest_switch_block_header) = self
+                        .storage
+                        .read_highest_switch_block_headers(1)
+                        .map_err(|err| err.to_string())?
+                        .last()
+                    {
+                        if !self.sync_to_genesis
```

Comment:
This is already checked above, probably just a copy-paste thing.

Reply:
Right, I added the optimization to potentially avoid calling the `read_highest_switch_block_headers()` [here](https://github.com/casper-network/casper-node/pull/3868/commits/e3a4e1fc33f6879667fedee821a94ff1e5f0be42), but forgot to remove the redundant, inner check. Good catch :+1: 

Fixed in 3ddf311830b0ae003e694f080a05ec973f062856